File: Patterns/Abstract Factory Pattern/CSharp/Pizza/VeggiePizza.cs
using System;

using PizzaIngredientFactory = Pizza.Ingredient.PizzaIngredientFactory;

namespace Pizza
{
    public class VeggiePizza : Pizza
    {
        private PizzaIngredientFactory ingredientFactory;

        public VeggiePizza(PizzaIngredientFactory ingredientFactory)
        {
            this.ingredientFactory = ingredientFactory;
        }

        public override void Prepare()
        {
            Console.WriteLine($"준비 중: {Name}");

            dough = ingredientFactory.CreateDough();
            sauce = ingredientFactory.CreateSauce();
            veggies = ingredientFactory.CreateVeggies();

            Console.WriteLine($"{dough?.Name}을 뿌리는 중...");
            Console.WriteLine($"{sauce?.Name}을 뿌리는 중...");
            Console.WriteLine("토핑을 올리는 중...");
            foreach(var veggie in veggies) {
                Console.WriteLine($"\t{veggie.Name}");
            }
        }
    }
}

File: Patterns/Abstract Factory Pattern/Cpp/Pizza/VeggiePizza.cpp
#pragma once

#include "./Pizza.cpp"
#include "./Ingredient/PizzaIngredientFactory.cpp"

class VeggiePizza : public Pizza {
public:
    PizzaIngredientFactory* ingredientFactory;

    VeggiePizza(PizzaIngredientFactory* ingredientFactory) : ingredientFactory{ingredientFactory} {}

    void prepare() override {
        cout << "준비 중: " << name;

        dough = ingredientFactory->createDough();
        sauce = ingredientFactory->createSauce();
        veggies = ingredientFactory->createVeggies();

        cout << dough->getName() << "를 돌리는 중...\n";
        cout << sauce->getName() << "를 뿌리는 중...\n";
        cout << "토핑을 올리는 중...\n";
        for (auto& i: veggies) {
            cout << '\t' << i->getName() << '\n';
        }
    }
};